zipline/algorithm.py

```diff
@@ -266,6 +266,7 @@ def __init__(self, *args, **kwargs):
         self.algoscript = kwargs.pop('script', None)
 
         self._initialize = None
+        self._analyze = kwargs.pop('analyze', None)
```

Comment:
> As far as a check for execution could I just pass a an analysis function that returns True as a test?

It looks like the consumer of `self._analyze` (in `analyze`) doesn't actually check the return value and will return anyway if `self._analyze` is unset.  

If `analyze` gets called in your test via `TradingAlgorithm.run`, then your analysis function could save a reference to the perf stats it's passed, and you could compare that it was passed the same object that was returned by `run`.  That seems to be the `analyze` interface.  Does that sound right to you?